src/observations/osculating.jl

```diff
@@ -0,0 +1,179 @@
+@doc raw"""
+    OsculatingElements{T <: AbstractFloat}
+
+Osculating orbital elements of a NEO. 
+
+# Fields 
+
+- `e::T`: eccentricity.
+- `q::T`: perihelion distance [au].
+- `tp::T`: time of pericenter passage [jd]. 
+- `Ω::T`: longitude of ascending node [deg].
+- `ω::T`: argument of pericentre [deg].
+- `i::T`: inclination [deg]. 
+- `a::T`: semimajor axis [au].
+"""
+@auto_hash_equals struct OsculatingElements{T <: AbstractFloat}
+    e::T
+    q::T
+    tp::T
+    Ω::T
+    ω::T
+    i::T
+    a::T
+    # Inner constructor 
+    function OsculatingElements{T}(e::T, q::T, tp::T, Ω::T, ω::T, i::T, a::T) where {T <: AbstractFloat}
+        return new{T}(e, q, tp, Ω, ω, i, a)
+    end
+end
+
+# Outer constructors
+function OsculatingElements(e::T, q::T, tp::T, Ω::T, ω::T, i::T, a::T) where {T <: AbstractFloat}
+    return OsculatingElements{T}(e, q, tp, Ω, ω, i, a)
+end
+
+function OsculatingElements()
+    return OsculatingElements(NaN, NaN, NaN, NaN, NaN, NaN, NaN)
+end
+
+# A OsculatingElements is NaN if all its fields are NaN 
+function isnan(osc::OsculatingElements{T}) where {T <: AbstractFloat} 
+    return isnan(osc.e) && isnan(osc.q) && isnan(osc.tp) && isnan(osc.Ω) && isnan(osc.ω) && isnan(osc.i) && isnan(osc.a)
+end
+
+# Print method for OsculatingElements
+# Example: 
+# Semimajor axis (a):                 0.8717319220347314 au
+# Eccentricity (e):                   0.4231715487782969
+# Time of pericenter passage (tp):    2021-03-19T03:45:01.293 JDTDB
+# Pericenter distance (q):            0.5028397744678126 au
+# Argument of pericenter (ω):         194.74654283451 deg
+# Inclination (i):                    34.81327005431841 deg
+# Longitude of Ascending Node (Ω):    17.855086873010706 deg
+function show(io::IO, m::OsculatingElements{T}) where {T <: AbstractFloat} 
+    
+    print(io, rpad("Semimajor axis (a): ", 36), m.a, " au\n")
+    print(io, rpad("Eccentricity (e): ", 36), m.e, "\n")
+    if isnan(m.tp)
+        print(io, rpad("Time of pericenter passage (tp): ", 36), "NaN JDTDB\n")
+    else 
+        print(io, rpad("Time of pericenter passage (tp): ", 36), julian2datetime(m.tp), " JDTDB\n")
+    end 
+    print(io, rpad("Pericenter distance (q): ", 36), m.q, " au\n")
+    print(io, rpad("Argument of pericenter (ω): ", 36), m.ω, " deg\n")
+    print(io, rpad("Inclination (i): ", 36), m.i, " deg\n")
+    print(io, rpad("Longitude of Ascending Node (Ω): ", 36), m.Ω, " deg\n")
```

Comment:
Perhaps, instead of `print` the output string (to `io`), we should use `@info`; the advantage of the latter is that it can be checked in tests. Do you think my suggestion is worth?

Reply:
This is the custom print method for `OsculatingElements`, so I think it is better to leave it as it is.